independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java

```diff
@@ -26,6 +27,49 @@
  */
 public final class RunnerClassLoader extends ClassLoader {
 
+    static class BufferView implements ClassLoadingResource.BufferSpan {
+
+        private static final BufferView EMPTY = new BufferView();
+        private byte[] data;
+        private int length;
+
+        // This is used to avoid "moving" an already freshly created buffer when
+        // pushing it back in the stack due to a null JarResource::getResourceData.
+        // if   null    ->  freshly new created one or EMPTY
+        // if   !null   ->  this has been already pushed to the stack: need to move its data
+        private BufferView next;
+
+        private BufferView(byte[] data, int length) {
+            this.data = data;
+            this.length = length;
+        }
+
+        private BufferView() {
+        }
+
+        @Override
+        public byte[] data() {
+            return data;
+        }
+
+        @Override
+        public int length() {
+            return length;
+        }
+
+        @Override
+        public BufferView moveToSpanWith(final int requiredCapacity) {
+            var data = this.data;
```

Comment:
Could we please avoid `var` ?
I understand `var`s are convenient to write, but for such pieces of code I value the ability to clearly understand what's happening more.

Reply:
Yep, I usually use it when the type is longer more than 3 chars and the name of the variable is "decent", but for github review is indeed bad, TBH and on the screen we have IDEA which tell which type is...let me replace them :+1: